test/flycheck-test.el

```diff
@@ -3911,11 +3911,37 @@ Why not:
     (add-hook 'scheme-mode-hook setup-geiser)
     (unwind-protect
         (flycheck-ert-should-syntax-check
-         "language/chicken.scm" 'geiser-mode
+         "language/chicken/error.scm" 'geiser-mode
          '(2 nil warning "in procedure call to `g1', expected a value of type `(procedure (* *) *)' but was given a value of type `number'"
              :checker scheme-chicken))
       (remove-hook 'scheme-mode-hook setup-geiser))))
 
+(flycheck-ert-def-checker-test scheme-chicken scheme syntax-error
+  (let ((setup-geiser
+         (lambda ()
+           (setq-local geiser-scheme-implementation 'chicken)
+           (geiser-mode))))
```

Comment:
I think we should reduce this duplication.  Please define:

```
(defun flycheck/chicken-mode ()
  "Enable Scheme and Geiser mode for Chicken scheme."
  (scheme-mode)
  (setq-local geiser-scheme-implementation 'chicken)
  (geiser-mode))
```

and use `'flycheck/chicken-mode` instead of `'geiser-mode` in all `flycheck-ert-should-syntax-check` invocations instead.